infrastructure/main.tf

```
@@ -1,19 +1,143 @@
-  default = "UK South"
+resource "azurerm_key_vault" "main" {
+  name                      = "kv-team3-jobapp-${var.environment}"
+  location                  = azurerm_resource_group.main.location
+  resource_group_name       = azurerm_resource_group.main.name
+  tenant_id                 = data.azurerm_client_config.current.tenant_id
+  sku_name                  = "standard"
+  enable_rbac_authorization = true
 }
 
-resource "azurerm_resource_group" "main" {
-  name     = "${var.app_name}-${var.environment}-rg"
-  location = var.region
+resource "azurerm_container_app_environment" "main" {
+  name                = "cae-${var.app_name}-${var.environment}"
+  location            = azurerm_resource_group.main.location
+  resource_group_name = azurerm_resource_group.main.name
+}
+
+resource "azurerm_user_assigned_identity" "frontend" {
+  name                = "mi-${var.app_name}-frontend-${var.environment}"
+  location            = azurerm_resource_group.main.location
+  resource_group_name = azurerm_resource_group.main.name
+}
+
+resource "azurerm_user_assigned_identity" "backend" {
+  name                = "mi-${var.app_name}-backend-${var.environment}"
+  location            = azurerm_resource_group.main.location
+  resource_group_name = azurerm_resource_group.main.name
+}
+
+resource "azurerm_role_assignment" "frontend_kv" {
+  scope              = azurerm_key_vault.main.id
+  role_definition_name = "Key Vault Secrets User"
+  principal_id       = azurerm_user_assigned_identity.frontend.principal_id
+}
+
+resource "azurerm_role_assignment" "frontend_acr" {
+  scope              = data.azurerm_container_registry.acr.id
+  role_definition_name = "AcrPull"
+  principal_id       = azurerm_user_assigned_identity.frontend.principal_id
+}
+
+resource "azurerm_role_assignment" "backend_kv" {
+  scope              = azurerm_key_vault.main.id
+  role_definition_name = "Key Vault Secrets User"
+  principal_id       = azurerm_user_assigned_identity.backend.principal_id
+}
+
+resource "azurerm_role_assignment" "backend_acr" {
+  scope              = data.azurerm_container_registry.acr.id
+  role_definition_name = "AcrPull"
+  principal_id       = azurerm_user_assigned_identity.backend.principal_id
 }
+
+resource "azurerm_container_app" "frontend" {
+  name                         = "ca-${var.app_name}-frontend-${var.environment}"
+  container_app_environment_id = azurerm_container_app_environment.main.id
+  resource_group_name          = azurerm_resource_group.main.name
+  revision_mode                = "Single"
+
+  identity {
+    type         = "UserAssigned"
+    identity_ids = [azurerm_user_assigned_identity.frontend.id]
+  }
+
+  template {
+    container {
+      name   = "frontend"
+      image  = var.frontend_image
+      cpu    = var.frontend_cpu
+      memory = var.frontend_memory
+    }
+
+    min_replicas = 1
+    max_replicas = 3
+  }
+
+  ingress {
+    allow_insecure_connections = false
+    external_enabled           = true
```

```suggestion
    external_enabled           = false
```
Since this is backend